python/MetrologyData.py

```diff
@@ -69,10 +71,26 @@ def xyzPlane_fit(self, nsigma=3, p0=(0, 0, 0)):
         dz = xyz_plane(positions, *pars) - self.z
         mean, stdev = np.mean(dz), np.std(dz)
 
-        # Refit the reference data within nsigma*stdev of the mean.
+        # Refit iteratively until the standard deviation of residuals does not
+        # change
+        stdev_last = -1
+        stdev_new = stdev
         index = np.where((dz > mean-nsigma*stdev) & (dz < mean+nsigma*stdev))
-        pars, _ = scipy.optimize.curve_fit(xyz_plane, positions[index],
-                                           self.z[index], p0=pars)
+        while stdev_new <> stdev_last:
```

Comment:
I already noted that this should use `!=` instead of `<>` in the PR for LSSTTD-859